### PR TITLE
*: fix pessimistic transaction and enable the test

### PR DESCRIPTION
**Pull request by coocood**
### What problem does this PR solve? 
Fix pessimistic transaction bug, enable the test.

### What is changed and how it works?
- move pessimistic test into its own test suite.
- Define `ErrWriteConflict` in `tikv` package.
- Define `ErrConflict` in `mocktikv` package.
- Remove the `[write conflict]` mark.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

### CODE REVIEW

File: store/mockstore/mocktikv/mvcc_leveldb.go [hunk 1]
```diff
@@ -651,18 +654,6 @@ func prewriteMutation(db *leveldb.DB, batch *leveldb.Batch, mutation *kvrpcpb.Mu
 		}
 	}
 
-	dec1 := valueDecoder{
```
Comment (tiancaiamao) [inline store/mockstore/mocktikv/mvcc_leveldb.go]: Why this code block is removed?

Comment (coocood) [reply store/mockstore/mocktikv/mvcc_leveldb.go]: It's duplicated block.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**